swarm/storage/ldbstore.go

```diff
@@ -310,18 +313,12 @@ func decodeData(addr Address, data []byte) (*chunk, error) {
 
 func (s *LDBStore) collectGarbage() error {
 
-	// the running param prevents duplicate gc from starting when one is already running
-	s.lock.Lock()
-	if s.gc.running {
-		s.lock.Unlock()
+	// prevent duplicate gc from starting when one is already running
+	if len(s.gc.runC) == 0 {
```

Comment:
this allows for a second collect garbage call to block.
* the first one checks length and finds non 0
* the second one checks length and finds non 0
* the first pops the lock
* the second one blocks on line 320

while its not a big problem, a select with default -> return is the correct one

Reply:
Nice catch. Thanks.